src/plugins/wizard/public/application/utils/get_top_nav_config.tsx

```diff
@@ -54,10 +57,28 @@ export const getTopNavConfig = (
     savedWizardVis,
     visualizationState,
     styleState,
-    hasUnappliedChanges,
+    metadataState,
+    dispatch,
   }: TopNavConfigParams,
   { history, toastNotifications, i18n: { Context: I18nContext } }: WizardServices
 ) => {
+  const hasUnappliedChanges = metadataState.editorState.hasUnappliedChanges;
+  const isEmpty = metadataState.editorState.isEmpty;
+  const hasChange = metadataState.editorState.hasChange;
+  const shouldRenderSaveButton = !isEmpty && hasChange && !hasUnappliedChanges;
```

Comment:
Rename this to `canSave`. You aren't preventing the rendering of the save button, you are just disabling it 😊